### ISSUE TITLE
Add support to remove externals from command line

### ISSUE THREAD

**Issue by alalazo** [LLNL/spack#19351]
As a user I want to be able to remove external specs in`packages.yaml` from the command line so that I don't have to manually edit the configuration.

### Rationale
With the addition of `spack external find` more and more packages are getting the ability to be discovered on the host system and be reused. While this is good, it also increases the size / complexity of the corresponding `packages.yaml` and sometimes manual editing is still necessary to fix an automatically detected configuration. 

For instance, the model that Spack implements for discovery is currently targeted at build tools and packages that may act both as a build tool and a library (based on their dependents) can potentially leave a broken configuration see e.g. the discussion at #19343. 

At the moment, if after:
```console
$ spack external find
[ ... ]
```
one or more of the packages being discovered needs to be removed, the user needs to manually edit the corresponding `packages.yaml` file. It would be more compact / convenient to have a way to do that from the command line based on specs.

### Description

The proposal is simply to add a subcommand:
```console
$ spack external remove <spec> [<spec> ...]
```
that removes any matching external from `packages.yaml`. It would be probably good to add an option to:
```console
$ spack external list --registered
```
to show which specs are currently in `packages.yaml`

### Additional information
--

### General information
- [x] I have searched the issues of this repo and believe this is not a duplicate

**Comment by johnwparent** [comment]
@alalazo Did this ever make it in? I can't find a PR to this effect nor any documentation for such a feature. I'd be _happy_ to add this myself as it's become quite relevant for Spack on Windows development lately.

**Comment by alalazo** [comment]
@johnwparent No, I added that as a feature as I saw a possible need for it, but the issue has been low priority ever since. If we want to add that let me know. I can either review a PR or try to find some time to implement it.

**Comment by johnwparent** [comment]
I'll see about finding a cycle or two to push on a PR to this effect, it'd be very useful for Spack on Windows. As I continually port packages and want to frequently add/remove a given package/deps for testing reasons it would be very convenient to not have to pull up a text editor and find the entry in the externals.
